scripts/generate_cli_api.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of REANA.
+# Copyright (C) 2020 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+"""REANA client CLI API docs generation."""
+
+import click
+
+from reana_client.cli import cli
+
+
+def _print_code_block(content):
+    print('```')
+    print(content)
+    print('```')
+
+
+print('# reana-client CLI API')
+with click.Context(cli) as ctx:
+    _print_code_block(cli.get_help(ctx))
+print('## Commands')
+for cmd_obj in cli.commands.values():
+    print('## {}'.format(cmd_obj.name))
```

Comment:
```suggestion
    print('### {}'.format(cmd_obj.name))
```

Otherwise, the commands do not appear nested.
![image](https://user-images.githubusercontent.com/9035606/79336963-f093f500-7f24-11ea-9e7c-d8afcf40ad63.png)

Reply:
More on nesting, what do you think of keeping the same logic separation we have in the help command inside the docs?
```
Usage:  [OPTIONS] COMMAND [ARGS]...

  REANA client for interacting with REANA server.

Options:
  -l, --loglevel [DEBUG|INFO|WARNING]
                                  Sets log level
  --help                          Show this message and exit.

Configuration commands:
  ping     Check connection to REANA server.
  version  Show version.

Workflow management commands:
  create  Create a new workflow.
  delete  Delete a workflow.
  diff    Show diff between two workflows.
  list    List all workflows and sessions.

Workflow execution commands:
  logs      Get workflow logs.
  restart   Restart previously run workflow.
  run       Shortcut to create, upload, start a new workflow.
  start     Start previously created workflow.
  status    Get status of a workflow.
  stop      Stop a running workflow.
  validate  Validate workflow specification file.

Workspace interactive commands:
  close  Close an interactive session.
  open   Open an interactive session inside the workspace.

Workspace file management commands:
  download  Download workspace files.
  du        Get workspace disk usage.
  ls        List workspace files.
  mv        Move files within workspace.
  rm        Delete files from workspace.
  upload    Upload files and directories to workspace.

Secret management commands:
  secrets-add     Add secrets from literal string or from file.
  secrets-delete  Delete user secrets by name.
  secrets-list    List user secrets.
```

So we would yield the following in markdown:
```markdown
# reana-client CLI API
## Configuration commands
### ping
...
### version
....
## Workflow management commands
### create
...
#### delete
....
```



We already have the logic to go through the click groups [here](https://github.com/reanahub/reana-client/blob/e2eb481775e8a0ae194ca2b66fa601c808246761/reana_client/cli/__init__.py#L58-L71).

Reply:
1st comment: Sure, actually I have it changed locally but I didn't push it.
2nd comment: OK, I see that's easy to navigate inside the groups:
```py
In [1]: from reana_client.cli import cli                                                                                                                                                                                                          

In [2]: cli.cmd_groups                                                                                                                                                                                                                            
Out[2]: 
[<Group configuration-group>,
 <Group workflow-management-group>,
 <Group workflow-execution-group>,
 <Group interactive-group>,
 <Group files-group>,
 <Group secrets-group>]

In [3]: cli.cmd_groups[0]                                                                                                                                                                                                                         
Out[3]: <Group configuration-group>

In [4]: cli.cmd_groups[0].commands                                                                                                                                                                                                                
Out[4]: {'ping': <Command ping>, 'version': <Command version>}
```